target/linux/ath79/dts/ar9331_glinet_gl-usb150.dts

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9331.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+
+/ {
+	compatible = "glinet,gl-usb150", "qca,ar9331";
+	model = "GL.iNet GL-USB150";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &eth0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "green:wlan";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led_power: power {
+			label = "green:power";
+			gpios = <&gpio 13 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_lan_reset {
+			gpio-export,name = "lan:reset";
+			gpio-export,output = <1>;
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
```

Comment:
for the `gpio_lan_reset ` 
I am not quite sure about this

in ar71xxx:
```
    gpio_request_one(GL_USB150_GPIO_LAN_RESET,
             GPIOF_OUT_INIT_LOW | GPIOF_EXPORT_DIR_FIXED,
             "LAN reset");
```

Reply:
If that pin is required to be pulled low for the USB-Ethernet Transceiver to work, it could probably be declared as a `reset-gpio` for the corresponding phy node (according to the bootlog this is `phy4` of `mdio1`), although it is technically not the phy itself that would react upon a reset; so, I'm not sure whether this would be acceptable anyways:

```
&swphy4 {
	reset-gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
}
```
It is active high here, since the driver will write the reset output to logic 0 ("not reset") for normal operation, resulting in a low level on the line (as it is currently achieved by writing 1 to an active low gpio).

Reply:
@s-2 thanks, but I never seem this
and should this be a  regulator?
```
    eth_vbus {
        compatible = "regulator-fixed";
        regulator-name = "eth_vbus";
        regulator-min-microvolt = <3300000>;
        regulator-max-microvolt = <3300000>;
        regulator-always-on;
        gpio = <&gpio 7 GPIO_ACTIVE_LOW>;
    };
```

Reply:
That would be another way to declare it, and probably matches the purpose of the pin even more closely. 

It is actually still an issue that certain devices require special pins to be set, while OpenWRT is trying to abandon the upstream-incompatible `gpio-export` (c.f. https://github.com/openwrt/openwrt/pull/1366#issuecomment-424834448).

However there seems to be no perfect replacement for all use cases yet, e.g. on DIR-510L we used `ucidef_add_gpio_switch` to enable USB power (as recommended in the PR), which unfortunately broke extroot (c.f. https://github.com/openwrt/openwrt/pull/3191#issuecomment-657658654).
Just like `gpio-export`, it allows the user change the pin state during runtime, but for this router it would probably break failsafe mode via USB.

In conclusion, I don't have any opinion on whether to use `reset-gpios`, `regulator-fixed` or `gpio-hog` (or deprecated `gpio-export`, allowing the user to disable USB), so let's wait to hear comments from maintainers.

Reply:
tested and yes it is GPIO_ACTIVE_HIGH
will change in next push.

Reply:
IIRC that last state was the author might choose between gpio-hog, gpio-export and user-space ucidef depending on what he/she preferred and what seemed reasonable for the device at hand.
I can comment on the benefits or drawbacks of regulator-fixed, however it seems like that one is used for the other glinet devices with ar9331 in ath79 target so far. ar9330 GL-AR150 uses a gpios property at &usb_phy instead.

So, it looks like there is room to choose.